pkg/sink/kafka/factory.go

```diff
@@ -115,6 +121,27 @@ func (p *saramaSyncProducer) SendMessages(_ context.Context, topic string, parti
 	return cerror.WrapError(cerror.ErrKafkaSendMessage, err)
 }
 
+func (p *saramaSyncProducer) HeartbeatBrokers() {
+	// We only send heartbeat to brokers when the last heartbeat time is
+	// older than the keep connection alive interval, to avoid sending heartbeat
+	// too frequently.
+	// This function will be called periodically in DDLSink.WriteCheckpointTs.
+	if time.Since(p.lastHeartbeatTime) < p.keepConnAliveInterval {
+		return
+	}
+	p.lastHeartbeatTime = time.Now()
+
+	// We don't care about the response and error here, even the connection
+	// is unestablished, we just need to keep the connection alive WHEN it's established.
+	// The connection will be established when a producer send messages.
+	// This is a workaround for the issue that sarama doesn't keep the connection alive
+	// when the connection is idle for a long time and we have disabled the retry in sarama.
+	brokers := p.client.Brokers()
+	for _, b := range brokers {
+		_, _ = b.Heartbeat(&sarama.HeartbeatRequest{})
+	}
```

Comment:
It is better to use a function to replace it.

Reply:
This code snippet repeats three times in your PR. If the code needs to be modified, we have to update it three times.